Whiskey/Functions/Invoke-WhiskeyTask.ps1

```diff
@@ -169,7 +169,11 @@ function Invoke-WhiskeyTask
 
     [hashtable]$taskProperties = $Parameter.Clone()
     $commonProperties = @{}
-    foreach( $commonPropertyName in @( 'OnlyBy', 'ExceptBy', 'OnlyOnBranch', 'ExceptOnBranch', 'OnlyDuring', 'ExceptDuring', 'WorkingDirectory', 'IfExists', 'UnlessExists', 'OnlyOnPlatform', 'ExceptOnPlatform' ) )
+    foreach( $commonPropertyName in @(
+        'OnlyBy', 'ExceptBy', 'OnlyOnBranch',

```

Comment:
Group by related parameters?

```
        'OnlyBy',         'ExceptBy',
        'OnlyOnBranch',   'ExceptOnBranch',
        'OnlyDuring',     'ExceptDuring',
        'OnlyOnPlatform', 'ExceptOnPlatform',
        'IfExists', 'UnlessExists',
        'WorkingDirectory', 'OutVariable'
```

Reply:
Don't do this now, but maybe instead of adding all these built-in parameters, these become tasks that run a list of tasks if their condition is met?

```
- RunTasks:
    InWorkingDirectory: 'fubar'
    By: BuildServer
    During: Build
    ExceptOnPlatform: macOS
    Tasks:
    - Log:
        Message: I'm in fubar:
```

`Invoke-WhiskeyTask` is already kind of hairy.

Reply:
I agree `Invoke-WhiskeyTask` is getting unwieldy but that doesn't seem like the right solution. Just looks like another way to call the `Pipeline` task.

Reply:
Maybe we move this function to `Pipeline` task and rename that task to `RunTasks`?

Reply:
I don't know, I'm still not convinced. What are we really trying to achieve by that? It seems like we'd be making the codebase a bit cleaner at the expense of the `whiskey.yml` syntax.